src/main/java/com/github/manolo8/darkbot/config/actions/values/StatTypeValue.java

```diff
@@ -0,0 +1,94 @@
+package com.github.manolo8.darkbot.config.actions.values;
+
+import java.util.Locale;
+
+import org.jetbrains.annotations.Nullable;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.config.actions.Parser;
+import com.github.manolo8.darkbot.config.actions.SyntaxException;
+import com.github.manolo8.darkbot.config.actions.Value;
+import com.github.manolo8.darkbot.config.actions.ValueData;
+import com.github.manolo8.darkbot.config.actions.parser.ParseUtil;
+
+import eu.darkbot.api.game.stats.Stats;
+import eu.darkbot.api.managers.StatsAPI;
+
+@ValueData(name = "stat-type", description = "Gets a certain Stat type from a bot", example = "stat-type(EXPERIENCE, earned)")
+public class StatTypeValue implements Value<Number>, Parser {
+    private StatsAPI.Key statKey;
+    private StatData dataType;
+
+    @Override
+    public @Nullable Number get(Main main) {
+        if (dataType == null || statKey == null) {
+            return null;
+        }
+
+        if (dataType == StatData.CURRENT) {
+            return main.statsManager.getStat(statKey).getCurrent();
+        } else if (dataType == StatData.EARNED) {
+            return main.statsManager.getStat(statKey).getEarned();
+        } else if (dataType == StatData.SPENT) {
+            return main.statsManager.getStat(statKey).getSpent();
+        }
+
+        return null;
+    }
+
+    private StatsAPI.Key getKeyFromString(String key) {
+        for (StatsAPI.Key stat : Stats.General.values()) {
+            if (stat.name().equals(key)) {
+                return stat;
+            }
+        }
+
+        return null;
+    }
+
+    public enum StatData {
+        CURRENT,
+        EARNED,
+        SPENT;
```

Comment:
Missing a "difference" or something like that, which is earned-spent, its what makes the most sense for graphs and overall "how much did the bot make"

Reply:
Wouldn't that be EARNED?

Reply:
Currently earned is only positive differences, so if you earn 100.000 uri, but spend 200.000 uri to gain it, stuff like uri/h will show positive, but the graph of overall uri should be going down (hence, difference, or earned - spent)